### PR TITLE
Update logging in Tipbox

### PR DESCRIPTION
**Pull request by mdp**
Big update on logging. Errors are still free of any information, they simply let us know that an error has occurred. For debugging you can drop to 'warn' level and get the full error.

Also now catches uncaught errors so that we can see catch anything catastrophic happening.

### CODE REVIEW

File: server/config/middlewares.js [hunk 1]
```diff
@@ -21,15 +22,16 @@ function getKeyRing(){
...
-            console.log('Err:', err);
+            logger.error('Error unlocking pgp');
+            logger.warn('Error unlocking pgp:', err);
             process.exit();
           }
 
         });
       } else {
-        console.log('Loaded private key w/o passphrase');
+        logger.prod('Loaded private key w/o passphrase');
```
Comment (xdamman) [inline server/config/middlewares.js:34]: why do you need to log that?

File: server/config/settings.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ var defaults = {
       'hostDomain': env.HOST_DOMAIN || env.MAILGUN_DOMAIN || 'tipbox.dev',
       'hmacKey': env.HMAC_KEY || '',
       'groupId': env.OC_GROUPID,
-      'apiKey': env.OC_API_KEY
+      'apiKey': env.OC_API_KEY,
+      'logLevel': 'warn'
```
Comment (xdamman) [inline server/config/settings.js]: shouldn't the default be "error"? Looks like "warn" log much more info.

File: server/config/middlewares.js (hunk 1 above)
Comment (mdp) [inline server/config/middlewares.js:34]: I think when in production we should default to showing security related config information on startup to help avoid configuration mistakes.

Things like, are we routing via Mailgun (less privacy), and are we loading a key that doesn't have a passphrase (probably a bad idea)

File: server/config/settings.js (hunk 2 above)
Comment (mdp) [inline server/config/settings.js]: Yeah, I'll change this.

File: server/config/middlewares.js (hunk 1 above)
Comment (xdamman) [inline server/config/middlewares.js:34]: Gotcha. Good idea.

**Pull request merged**